server/src/main/java/org/elasticsearch/index/query/AbstractGeometryQueryBuilder.java

```diff
@@ -341,7 +341,8 @@ public boolean ignoreUnmapped() {
     /** builds the appropriate lucene shape query */
     protected abstract Query buildShapeQuery(QueryShardContext context, MappedFieldType fieldType);
     /** returns expected content type for this query */
-    protected abstract String queryFieldType();
+    @Deprecated
+    protected abstract List queryFieldType();
```

Comment:
You cannot change the signature of the method and deprecate it. In any case I think this an internal API so we probably can just remove the method all together (I will check).

Reply:
It is only used in the place where I have replaced it so I will remove for simplicity

Reply:
It used used as well in the XPack spatial module, you need to reflect these changes there as well.

Reply:
Thanks, that's fixed now, I hadn't grepped through X-Pack